### ISSUE TITLE
Clear output console when reloading scene

### ISSUE THREAD

**Issue by acgc99** [godotengine/godot-proposals#8153]
### Describe the project you are working on

A [plugin](https://github.com/acgc99/Godot-Widgets) with widgets for mobile apps.

### Describe the problem or limitation you are having in your project

I'm using many `@tool` scripts, sometimes they have a print or there is an error, so messages appear on the console. Then, I modify the code and press `Scene > Reload Saved Scene` and the printted messages from previous code are still there, although scene is running according to the new code.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Cleaning the console when scene is reloaded will help debugging `@tool` scripts.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When `Scene > Reload Saved Scene` is pressed, the console is cleaned.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so.

### Is there a reason why this should be core and not an add-on in the asset library?

Yes, I think this is something basic and useful. I don't think any reason why not cleaning it is an advantage.

**Comment by AThousandShips** [comment]
This isn't how the console is used by most people, most people don't treat it as specific to a scene I'd say, and it's trivial to clear it manually so I don't see the usefulness of this

Unlike clearing the console when running the game, which is optional, there's no real boundary when loading a scene that makes sense to me as a place where you'd want to clear the console

**Comment by Calinou** [comment]
I think it makes sense to display a separator in the Output panel when reloading the scene, but I wouldn't clear it entirely.

**Comment by acgc99** [comment]
> In fact, I'd argue this would also be preferable behavior when running the project. Separators are nondestructive, while clearing the console is. **Edit:** This appears to already be the case thanks to the "Debugging process started/stopped" messages when **Always Clear Output on Play** is disabled:

That is true if you are running the project, but if you do `Scene > Reload Saved Scene`, `Debugging process started/stopped` is not printed in that case (makes sense). Remember I'm working with `@tool` scripts, so I don't even press the play button.

Following a similar approach I would suggest to print an information message when `Scene > Reload Saved Scene` is pressed, or even when a scene is loaded in the Editor, so we know from which scene below messages come. For example `Scene X (re)loaded`.
